### ISSUE TITLE
Multiple targets in mappings source not working?

### ISSUE THREAD

**Issue by leetal** [cts2/lexevs-service#15]
Hi again! 

Sorry for a long post.

**TL;DR:** Multi-target mappings not working as expected. Example files in bottom of post. Help needed ;)

I have during the last week been trying to resolve multiple target codes in a mapping scheme, but due to some unknown reason i always get ONE result back, despite if i have multiple targets in a mapping source->target(s) association like so:

**int_to_sodertalje_mapping.xml**
```xml
<relations representsVersion="1.0" containerName="mappings" isMapping="true"
        sourceCodingScheme="internal" sourceCodingSchemeVersion="1.0"
        targetCodingScheme="sodertalje" targetCodingSchemeVersion="1.0">
  <lgRel:associationPredicate associationName="mapsTo">
    <lgRel:source sourceEntityCodeNamespace="internal" sourceEntityCode="B13100">
      <lgRel:target targetEntityCodeNamespace="sodertalje" targetEntityCode="13000" />
      <lgRel:target targetEntityCodeNamespace="sodertalje" targetEntityCode="13300" />
    </lgRel:source>
...
</relations>
```

**What i want:**
Code _B13100_ should return a mapping to _13000_ **AND** _13300_. I only get **ONE** mapping to _13000_ as can be seen below after querying _B13100_.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<MapEntryMsg xmlns="http://www.omg.org/spec/CTS2/1.1/MapVersion"
    xmlns:core="http://www.omg.org/spec/CTS2/1.1/Core"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.omg.org/spec/CTS2/1.1/MapVersion http://informatics.mayo.edu/cts2/spec/CTS2/1.1/mapversion/MapVersion.xsd">
    <core:heading>
        <core:resourceRoot>http://HIDDENURL/cts2/</core:resourceRoot>
        <core:resourceURI>map/internal_sodertalje/version/internal_sodertalje-1.0/entry/internal:B13100</core:resourceURI>
        <core:accessDate>2016-10-25T10:41:18.564+02:00</core:accessDate>
    </core:heading>
    <entry entryState="ACTIVE" processingRule="ALL_MATCHES">
        <assertedBy>
            <core:mapVersion href="http://HIDDENURL/cts2/codesystem/internal_sodertalje/version/1.0">internal_sodertalje-1.0</core:mapVersion>
            <core:map href="http://HIDDENURL/cts2/map/internal_sodertalje">internal_sodertalje</core:map>
        </assertedBy>
        <mapFrom uri="urn:bft:internal:internal:B13100">
            <core:namespace>internal</core:namespace>
            <core:name>B13100</core:name>
        </mapFrom>
        <mapSet processingRule="ALL_MATCHES" entryOrder="1">
            <mapTarget entryOrder="1">
                <mapTo uri="urn:bft:external:sodertalje:13000">
                    <core:namespace>sodertalje</core:namespace>
                    <core:name>13000</core:name>
                </mapTo>
            </mapTarget>
        </mapSet>
    </entry>
</MapEntryMsg>
```

**Problem:**
I have tried lots of combinations, but all give the same result... This is in close conjunction with #8 but with a complete set of data.

What is interesting (and also making me believe that i have done something wrong) is that if i try to query the map URL `http://{URL}/cts2/map/internal_sodertalje/version/internal_sodertalje-1.0/entries`, it will actually display mutiple entries of the same code (the exact same amount as there is targets to the given source in the mappings file, which to me seems inverted somehow).

Below is part of the output to better clarify what i mean: 
```xml
<?xml version="1.0" encoding="UTF-8"?>
<MapEntryDirectory xmlns="http://www.omg.org/spec/CTS2/1.1/MapVersion"
    xmlns:core="http://www.omg.org/spec/CTS2/1.1/Core"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://www.omg.org/spec/CTS2/1.1/MapVersion http://informatics.mayo.edu/cts2/spec/CTS2/1.1/mapversion/MapVersion.xsd"
    complete="PARTIAL" numEntries="50" next="http://HIDDENURL/cts2/map/internal_sodertalje/version/internal_sodertalje-1.0/entries?page=1&amp;maxtoreturn=50">
    <core:heading>
        <core:resourceRoot>http://HIDDENURL/cts2/</core:resourceRoot>
        <core:resourceURI>map/internal_sodertalje/version/internal_sodertalje-1.0/entries</core:resourceURI>
        <core:accessDate>2016-10-25T10:17:22.505+02:00</core:accessDate>
    </core:heading>
    <entry
        href="http://HIDDENURL/cts2/map/internal_sodertalje/version/internal_sodertalje-1.0/entry/internal:B13100" resourceName="internal:B13100">
        <assertedBy>
            <core:mapVersion href="http://HIDDENURL/cts2/codesystem/internal_sodertalje/version/1.0">internal_sodertalje-1.0</core:mapVersion>
            <core:map href="http://HIDDENURL/cts2/map/internal_sodertalje">internal_sodertalje</core:map>
        </assertedBy>
        <mapFrom uri="urn:bft:internal:internal:B13100">
            <core:namespace>internal</core:namespace>
            <core:name>B13100</core:name>
        </mapFrom>
    </entry>
    <entry
        href="http://HIDDENURL/cts2/map/internal_sodertalje/version/internal_sodertalje-1.0/entry/internal:B13100" resourceName="internal:B13100">
        <assertedBy>
            <core:mapVersion href="http://HIDDENURL/cts2/codesystem/internal_sodertalje/version/1.0">internal_sodertalje-1.0</core:mapVersion>
            <core:map href="http://HIDDENURL/cts2/map/internal_sodertalje">internal_sodertalje</core:map>
        </assertedBy>
        <mapFrom uri="urn:bft:internal:internal:B13100">
            <core:namespace>internal</core:namespace>
            <core:name>B13100</core:name>
        </mapFrom>
    </entry>
    <entry
        href="HIDDENURL/cts2/map/internal_sodertalje/version/internal_sodertalje-1.0/entry/internal:B30400" resourceName="internal:B30400">
        <assertedBy>
            <core:mapVersion href="HIDDENURL/cts2/codesystem/internal_sodertalje/version/1.0">internal_sodertalje-1.0</core:mapVersion>
            <core:map href="HIDDENURL/cts2/map/internal_sodertalje">internal_sodertalje</core:map>
        </assertedBy>
        <mapFrom uri="urn:bft:internal:internal:B30400">
            <core:namespace>internal</core:namespace>
            <core:name>B30400</core:name>
        </mapFrom>
    </entry>
...
</MapEntryDirectory>
```

**Example files:**
I have attached complete files that are importable into LexEVS that demonstrate the problem.
Here is the file: [example.zip](https://github.com/cts2/lexevs-service/files/550015/example.zip)
The archive consists of 3 LexGrid XML files, and the one generating the output above is the "int_to_sodertalje_mapping.xml" file, where there are multiple targets for almost every source defined in a 1-to-N relationship kind of way.

I cannot seem to find a working example of this kind of relationship anywhere. So if you could either give a short example, or actually just point out where my error is (or if the LexEVS-Service actually even can handle this kind of mapping), i would be very (very, very) happy! 

Regards
Alexander

**Comment by kevinpeterson** [comment]
In CTS2, a ```MapEntry``` is identified by its ```mapFrom``` entity URI.

So the XML that the service is returning:
```xml
...
    <entry
        href="http://HIDDENURL/cts2/map/internal_sodertalje/version/internal_sodertalje-1.0/entry/internal:B13100" resourceName="internal:B13100">
        <assertedBy>
            <core:mapVersion href="http://HIDDENURL/cts2/codesystem/internal_sodertalje/version/1.0">internal_sodertalje-1.0</core:mapVersion>
            <core:map href="http://HIDDENURL/cts2/map/internal_sodertalje">internal_sodertalje</core:map>
        </assertedBy>
        <mapFrom uri="urn:bft:internal:internal:B13100">
            <core:namespace>internal</core:namespace>
            <core:name>B13100</core:name>
        </mapFrom>
    </entry>
    <entry
        href="http://HIDDENURL/cts2/map/internal_sodertalje/version/internal_sodertalje-1.0/entry/internal:B13100" resourceName="internal:B13100">
        <assertedBy>
            <core:mapVersion href="http://HIDDENURL/cts2/codesystem/internal_sodertalje/version/1.0">internal_sodertalje-1.0</core:mapVersion>
            <core:map href="http://HIDDENURL/cts2/map/internal_sodertalje">internal_sodertalje</core:map>
        </assertedBy>
        <mapFrom uri="urn:bft:internal:internal:B13100">
            <core:namespace>internal</core:namespace>
            <core:name>B13100</core:name>
        </mapFrom>
    </entry>
...
```

is technically invalid. If there is a one-to-many mapping, the service should display it as one ```MapEntry``` with multiple targets. What you're trying to do is valid -- this is a code defect.
